src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -107,7 +111,78 @@ def _convert_to_assistant_definition(self, assistant_definition_path: str, input
         file = self._working_dir / assistant_definition_path
         with open(file, "r", encoding="utf-8") as file:
             assistant_definition = load_yaml(file)
-        return AssistantDefinition.deserialize(assistant_definition)
+        assistant_def = AssistantDefinition.deserialize(assistant_definition)
+        self._resolve_assistant_tool(assistant_def)
+        return assistant_def
+
+    def _resolve_assistant_tool(self, assistant_definition: AssistantDefinition):
+        resolved_tools = {}
+        for tool in assistant_definition.tools:
+            if tool["type"] in ("code_interpreter", "retrieval"):
+                resolved_tools[tool["type"]] = AssistantTool(name=tool["type"], openai_definition=tool, func=None)
+            elif tool["type"] == "function":
+                function_tool = self._load_tool_as_function(tool)
+                resolved_tools[function_tool.name] = function_tool
+            else:
+                raise UnsupportedAssistantToolType(
+                    message_format="Unsupported assistant tool type: {tool_type}",
+                    tool_type=tool["type"],
+                    target=ErrorTarget.EXECUTOR,
+                )
+        assistant_definition.tool_invoker = AssistantToolInvoker(resolved_tools)
+
+    def _load_tool_as_function(self, tool: dict):
+        node, predefined_inputs = self._create_node_for_assistant_tool(tool)
```

Comment:
We need to add a TODO here to decouple tool resolving and node resolving.

Reply:
We should also add more comments here to explain why we convert it to a node.

Reply:
https://msdata.visualstudio.com/Vienna/_workitems/edit/2979976